src/core/lib/slice/slice_buffer.cc

```diff
@@ -166,12 +167,27 @@ void grpc_slice_buffer_pop(grpc_slice_buffer* sb) {
 void grpc_slice_buffer_reset_and_unref_internal(grpc_exec_ctx* exec_ctx,
                                                 grpc_slice_buffer* sb) {
   size_t i;
-  for (i = 0; i < sb->count; i++) {
+  for (i = sb->idx_of_first_valid_slice; i < sb->count; i++) {
     grpc_slice_unref_internal(exec_ctx, sb->slices[i]);
   }
 
   sb->count = 0;
   sb->length = 0;
+  sb->idx_of_first_valid_slice = 0;
+}
+
+void grpc_slice_buffer_partial_unref_internal(grpc_exec_ctx* exec_ctx,
```

Comment:
Looks like we can achieve the same with:
for i 0 to idx:
grpc_slice s = grpc_slice_buffer_take_first(b);
grpc_slice_unref(s);
But you may not need it as mentioned above...

Reply:
Ack